universe/ShipDesign.cpp

```diff
@@ -675,9 +675,10 @@ void ShipDesign::BuildStatCaches() {
         { m_num_part_classes[part_class]++; }
     }
 
-    // ensure tags are unique and copy into this->m_tags
+    // ensure tags are unique
     std::sort(tags.begin(), tags.end());
     auto last = std::unique(tags.begin(), tags.end());
+    tags.erase(last, tags.end());
```

Comment:
```suggestion
```

Reply:
can't add a suggestion past line 685, but unsted something like:
````
    std::size_t tags_sz = 0;
    std::for_each(tags.begin(), last, [this](auto str) { tags_sz += str.size(); });
    m_tags_concatenated.reserve(tags_sz);
    m_tags.clear();
    m_tags.reserve(tags.size());
    std::for_each(tags.begin(), last, [this](auto str) {
        auto next_start = m_tags_concatenated.size();
    m_tags_concatenated.append(str);
````

Reply:
I added the above implementation, big thanks!